src/snowflake/snowpark/_internal/server_connection.py

```diff
@@ -695,24 +695,28 @@ def _fix_pandas_df_integer(
         if (
             FIELD_ID_TO_NAME.get(column_metadata.type_code) == "FIXED"
             and column_metadata.precision is not None
-            and column_metadata.scale == 0
             and not str(pandas_dtype).startswith("int")
```

Comment:
we are missing this check here. Is it possible that pandas_dtype is already "float64" or "float32" and we may process that column again. I do not think that is desirable. Can this be moved to the if condition below for making our distinction between floats and int

Reply:
Added the check for float.

Reply:
I would say something like

```
            if column_metadata.scale == 0 and not str(pandas_dtype).startswith("int"):
                # fix int
            elif column_metadata.scale > 0 and not str(pandas_dtype).startswith("float"):
                # fix float
```

Reply:
I see. That makes sense to me. 
@sfc-gh-sfan would you mind including this change as well, since you're working on it? Thanks!